### ISSUE TITLE
Pop-up is not dissmissed when clicking in a modal "curtain".

### ISSUE THREAD

**Issue by rodrigorc** [ocornut/imgui#7654]
### Version/Branch of Dear ImGui:

1.90.7

### Back-ends:

all

### Compiler, OS:

any

### Details:

1. Open a modal pop-up, and then over it open a normal pop-up, such as the "Combo" or "Color picker" in the "Stacked modals" demo.
2. When clicking outside of the normal pop-up it should be dismissed, but this only works if I click inside the modal window. If instead I click in the modal dimmed area nothing happens.

I can imagine a situation where a big transient pop-up could block the UI and the user would be unable to dismiss it.

### Screenshots/Video:

![image](https://github.com/ocornut/imgui/assets/1128630/b9676d70-237d-417a-8244-8d20e449c19d)

**Comment by ocornut** [comment]
Should be fixed by 0561d70.

- I was first puzzled that the "window_popup_menu" test was breaking but I realized the breakage was specifically under a `#if !IMGUI_BROKEN_TESTS` block aka was testing an undesirable behavior, which we now fixed.
- Then "window_popup_nested_interruptions" was breaking but it only highlighted that my first fix attempt was broken, at it accidentally allowed to close a nested modal by focusing the previous one. The correct fix is to call `ClosePopupsOverWindow(GetTopMostPopupModal(), ...`) and not `ClosePopupsOverWindow(blocking_modal);`.

Those specifics tests are a bit laborious to work but they are always helpful.
